### PR TITLE
Release 2.1.0

### PR DESCRIPTION
**Pull request by EvansPie**
Release 2.1.0

# What this PR does

- DEX-1968 Removed the flow and replaced it with paymentMethodType and intent mechanism
- DEX-1946 Fix Klarna checkout issue
- HUC is now part of the main SDK
- Minor codebase improvements

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckout.swift [hunk 1]
```diff
@@ -18,19 +18,15 @@ public class PrimerHeadlessUniversalCheckout {
     fileprivate init() {}
     
     public func start(withClientToken clientToken: String, settings: PrimerSettings? = nil, delegate: PrimerHeadlessUniversalCheckoutDelegate? = nil, completion: @escaping (_ paymentMethodTypes: [PrimerPaymentMethodType]?, _ err: Error?) -> Void) {
+        Primer.shared.intent = .checkout
```
Comment (d4r1091) [inline Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckout.swift:21]: is this always needed to be `.checkout` here?

File: Sources/PrimerSDK/Classes/Core/Primer/Primer.swift [hunk 2]
```diff
@@ -92,14 +92,15 @@ public class Primer {
      */
 
     public func showUniversalCheckout(clientToken: String, completion: ((Error?) -> Void)? = nil) {
+        intent = .checkout
```
Comment (d4r1091) [inline Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:95]: is this always needed to be `.checkout` here?

File: Sources/PrimerSDK/Classes/Core/Primer/Primer.swift [hunk 3]
```diff
@@ -174,25 +176,19 @@ public class Primer {
                     .buckarooSofort,
                     .coinbase,
                     .hoolah,
+                    .klarna,
```
Comment (d4r1091) [inline Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:179]: do we need `.klarna` among these payment method cases? asking as I've seen `.klarna` below (in the `else` condition) as well

File: Sources/PrimerSDK/Classes/Core/Primer/Primer.swift [hunk 4]
```diff
@@ -165,6 +166,7 @@ public class Primer {
                     .adyenTwint,
                     .adyenVipps,
                     .adyenPayshop,
+                    .applePay,
```
Comment (d4r1091) [inline Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:169]: do we need `.applePay` among these payment method cases?

File: Sources/PrimerSDK/Classes/Data Models/ClientSession.swift [hunk 5]
```diff
@@ -336,7 +336,7 @@ extension ClientSessionAPIResponse.Action {
     static func unselectPaymentMethodIfNeeded() -> Promise<Void> {
         return Promise { seal in
             
-            guard !Primer.shared.flow.internalSessionFlow.vaulted else {
+            guard Primer.shared.intent == .checkout else {
```
Comment (d4r1091) [inline Sources/PrimerSDK/Classes/Data Models/ClientSession.swift:339]: 👌

File: Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckout.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Core/Checkout Components/PrimerHeadlessUniversalCheckout.swift:21]: Yes

File: Sources/PrimerSDK/Classes/Core/Primer/Primer.swift (hunk 4 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Core/Primer/Primer.swift:169]: Yes, apple pay is only for checking out

**Review by d4r1091 — APPROVED** [review]

**Review by xevious78 — APPROVED** [review]

**Pull request merged**